src/style.rs

```diff
@@ -311,6 +313,18 @@ impl Style {
                     }
                 },
                 Some(49) => background = None,
+                Some(58) => match (parts.pop_front(), parts.pop_front()) {
+                    (Some(5), Some(color)) => underline = Some(Color::Fixed(color)),
+                    (Some(2), Some(red)) => match (parts.pop_front(), parts.pop_front()) {
+                        (Some(green), Some(blue)) => underline = Some(Color::RGB(red, green, blue)),
+                        _ => {
+                            break;
+                        }
+                    },
+                    _ => {
+                        break;
+                    }
+                },
```

Comment:
According to https://en.wikipedia.org/wiki/ANSI_escape_code, `58` is a non-standard addition. But it seems to be supported in some terminals, so I'm okay with adding it. Should we support `59` as well (default underline color)? Similar to, say, `24` for disabling underlining.

Reply:
Agreed. I just added `59`. Thanks.